### DISCUSSION TITLE
Is there any way that I want to remove duplicates in a array but maintaining the original relative order in array using cuda library? That means not sorting.

### DISCUSSION THREAD

**Discussion by zlwu92** [NVIDIA/cccl#1065] [CUB]
Moreover, if I have many of these arrays, and need to do the same thing for each array individually, can cub library help me to achieve this?

**Comment by gevtushenko** [comment]
Hello @zlwu92! Please, take a look at our [cub::DeviceSelect::Unique](https://github.com/NVIDIA/cub/blob/main/examples/device/example_device_select_unique.cu). 
Note, however, that our API considers unique elements in "runs" of keys. For instance:
```
in = [0, 0, 0, 0, 0, 0, 0, 0, 1, 1]
out = [0, 1] 
```  
but 
```
in = [0, 0, 0, 0, 0, 0, 0, 0, 1, 0]
out = [0, 1, 0] 
```  

Regarding your second question, we don't have segmented nor batched version of `cub::DeviceSelect::Unique` at the moment.
